Build fails with llvm-strip: not found

When i run `make minimal` on Ubuntu (20.04 & 22.04) it fails with  `/bin/sh: 1: llvm-strip: not found`. In order to get this working i followed the installation https://apt.llvm.org/, and then created a symbolic link `sudo ln -s /usr/bin/llvm-strip-12 /usr/bin/llvm-strip`. Can you please improve the README.md so that others don't get hung up on this.  

`vagrant@devenv:~/host_user_homedir/workspace/src/libbpf-bootstrap/examples/c$ make minimal
  MKDIR    .output
  MKDIR    .output/libbpf
  LIB      libbpf.a
.....
  INSTALL  /home/vagrant/host_user_homedir/workspace/src/libbpf-bootstrap/examples/c/.output//libbpf/libbpf.a
  BPF      .output/minimal.bpf.o
/bin/sh: 1: llvm-strip: not found
make: *** [Makefile:112: .output/minimal.bpf.o] Error 127
make: *** Deleting file '.output/minimal.bpf.o'`

--- a/examples/c/Makefile
+++ b/examples/c/Makefile
@@ -1,7 +1,6 @@
 # SPDX-License-Identifier: (LGPL-2.1 OR BSD-2-Clause)
 OUTPUT := .output
 CLANG ?= clang
-LLVM_STRIP ?= llvm-strip
 LIBBPF_SRC := $(abspath ../../libbpf/src)
 BPFTOOL_SRC := $(abspath ../../bpftool/src)
 LIBBPF_OBJ := $(abspath $(OUTPUT)/libbpf.a)
@@ -107,10 +106,11 @@ $(LIBBLAZESYM_HEADER): $(LIBBLAZESYM_SRC)/target/release/libblazesym.a | $(OUTPU
 	$(Q)cp $(LIBBLAZESYM_SRC)/target/release/blazesym.h $@
 
 # Build BPF code
-$(OUTPUT)/%.bpf.o: %.bpf.c $(LIBBPF_OBJ) $(wildcard %.h) $(VMLINUX) | $(OUTPUT)
+$(OUTPUT)/%.bpf.o: %.bpf.c $(LIBBPF_OBJ) $(wildcard %.h) $(VMLINUX) | $(OUTPUT) $(BPFTOOL)
 	$(call msg,BPF,$@)
-	$(Q)$(CLANG) -g -O2 -target bpf -D__TARGET_ARCH_$(ARCH) $(INCLUDES) $(CLANG_BPF_SYS_INCLUDES) -c $(filter %.c,$^) -o $@
-	$(Q)$(LLVM_STRIP) -g $@ # strip useless DWARF info
+	$(eval TMP_BPF_O := $(patsubst %.bpf.o,%.tmp.bpf.o,$@))
+	$(Q)$(CLANG) -g -O2 -target bpf -D__TARGET_ARCH_$(ARCH) $(INCLUDES) $(CLANG_BPF_SYS_INCLUDES) -c $(filter %.c,$^) -o $(TMP_BPF_O)
+	$(Q)$(BPFTOOL) gen object $@ $(TMP_BPF_O)
 
 # Generate BPF skeletons
 $(OUTPUT)/%.skel.h: $(OUTPUT)/%.bpf.o | $(OUTPUT) $(BPFTOOL)

